standard/date_ext.lua

```diff
@@ -52,7 +52,10 @@ function DateExt.readTimestamp(dateInput)
 
 	-- Extracts the '-4:00' out of <abbr data-tz="-4:00" title="Eastern Daylight Time (UTC-4)">EDT</abbr>
 	local tzTemplateOffset = dateInput:match('data%-tz%=[\"\']([%d%-%+%:]+)[\"\']')
-	local datePart = (mw.text.split(dateInput, '<', true)[1]):gsub('-', '')
+	local datePart = (mw.text.split(dateInput, '<', true)[1])
+		:gsub('-', '')
+		:gsub('T', '') -- is used as sep in forms datepicker if time is entered
+		:gsub('Z$', '') -- gets appended in forms datepicker if time is entered
```

Comment:
From a bit of testing, it seems like these aren't needed at all?
formatDate is capable of parsing these timestamps, it only "breaks" (returns plain wrong) when substing the dashes away
![image](https://github.com/user-attachments/assets/ca661788-45b6-4e70-b209-27bbe143108b)

Reply:
- `2024-05-26T11:29:54.000Z` plain errors with the current setup
- you are right can drop the trimming away of the Z though